livre/eq/Channel.cpp

```diff
@@ -258,8 +298,10 @@ struct Channel::Impl
         renderPipeline.render( pipe->getFrameData()->getVRParameters(),
                                _frameInfo,
                                {{ _drawRange.start, _drawRange.end }},
-                               PixelViewport( vp.x, vp.y, vp.w, vp.h ),
+                               PixelViewport( pixVp.x, pixVp.y, pixVp.w, pixVp.h ),
+                               Viewport( vp.x, vp.y, vp.w, vp.h ),
                                PipeFilterT< RedrawFilter >( "RedrawFilter", _channel ),
+                               PipeFilterT< SendHistogramFilter >( "SendHistogramFilter", _channel ),
                                *_renderer,
                                _frameInfo.nAvailable,
                                _frameInfo.nNotAvailable );
```

Comment:
You are already passing _frameInfo, why do you need to pass these two parameters separately.

Instead of passing the two viewports and the draw range as parameters I would simply pass the channel.
Then you can get FrameData from the pipe if you call channel->getPipe() in the implementation and the same goes for the frustum, which means that you can completely get rid of _frameInfo and pass the _frameCounter instead.
For nAvailable and nNotAvailable I would use a return value (struct or pair) because updating these variables from a reference inside a function that also has a const& to them in other parameter is pointing to a design flaw in my opinion.